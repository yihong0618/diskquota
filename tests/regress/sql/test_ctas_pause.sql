CREATE SCHEMA hardlimit_s;
SET search_path TO hardlimit_s;

\! gpconfig -c "diskquota.hard_limit" -v "on" > /dev/null
\! gpstop -u > /dev/null
SELECT diskquota.set_schema_quota('hardlimit_s', '1 MB');
SELECT diskquota.wait_for_worker_new_epoch();

-- heap table
CREATE TABLE t1 (i) AS SELECT generate_series(1,10000000) DISTRIBUTED BY (i); -- expect fail

SELECT diskquota.pause();

CREATE TABLE t1 (i) AS SELECT generate_series(1,10000000) DISTRIBUTED BY (i); -- expect succeed

-- disable hardlimit and do some clean-ups.
\! gpconfig -c "diskquota.hard_limit" -v "off" > /dev/null
\! gpstop -u > /dev/null
SELECT diskquota.resume();

DROP SCHEMA hardlimit_s CASCADE;
