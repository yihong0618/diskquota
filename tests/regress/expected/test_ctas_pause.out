CREATE SCHEMA hardlimit_s;
SET search_path TO hardlimit_s;
\! gpconfig -c "diskquota.hard_limit" -v "on" > /dev/null
\! gpstop -u > /dev/null
SELECT diskquota.set_schema_quota('hardlimit_s', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

-- heap table
CREATE TABLE t1 (i) AS SELECT generate_series(1,5000000) DISTRIBUTED BY (i); -- expect fail
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'generate_series' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ERROR:  schema's disk space quota exceeded with name:110528  (seg1 127.0.0.1:6003 pid=73892)
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

CREATE TABLE t1 (i) AS SELECT generate_series(1,5000000) DISTRIBUTED BY (i); -- expect succeed
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'generate_series' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
-- disable hardlimit and do some clean-ups.
\! gpconfig -c "diskquota.hard_limit" -v "off" > /dev/null
\! gpstop -u > /dev/null
SELECT diskquota.resume();
 resume 
--------
 
(1 row)

DROP SCHEMA hardlimit_s CASCADE;
NOTICE:  drop cascades to table t1
